Annex60/BoundaryConditions/SolarIrradiation/BaseClasses/SkyClearness.mo

```diff
@@ -18,34 +18,60 @@ block SkyClearness "Sky clearness"
   Modelica.Blocks.Interfaces.RealOutput skyCle
     "Sky clearness. skyCle=1: overast sky; skyCle=8: clear sky"
     annotation (Placement(transformation(extent={{100,-10},{120,10}})));
+  // Set hSmall so that hSmall + deltaX < 1E-4. See info section.
 protected
+  constant Modelica.SIunits.Irradiance hSmall = 0.5e-4
+    "Small radiation for regularization";
+  constant Modelica.SIunits.Irradiance deltaX = hSmall/2
+    "Small radiation for regularization";
   constant Real k = 5.534e-6*(180/Modelica.Constants.pi)^3 "Constant factor";
   Real tmp1 "Intermediate variable";
+  Modelica.SIunits.Irradiance HDifHorBou
+    "Diffuse horizontal irradiation, bounded away from zero";
 equation
   tmp1 =  k*zen^3;
+  HDifHorBou = Annex60.Utilities.Math.Functions.smoothMax(
+                 x1 = HDifHor,
+                 x2 = hSmall,
+                 deltaX = deltaX);
+  // In the Buildings library, HGloHor is always larger than 1E-4
+  // (minus some small undershoot due to regularization. Hence,
+  // it makes no sense to simplify the equation for
+  // HGloHor < Modelica.Constants.small.
+  skyCle = Annex60.Utilities.Math.Functions.smoothLimit(
+        x = (HGloHor/HDifHorBou + tmp1)/(1 + tmp1),
+        l = 1,
+        u = 8,
+        deltaX = 0.01);
 
-  skyCle =  smooth(1,
-    if (HGloHor < Modelica.Constants.small)
-      then
-        1
-      else
-       Annex60.Utilities.Math.Functions.smoothLimit(
-        x=  (HGloHor/Annex60.Utilities.Math.Functions.smoothMax(
-                       x1=  HDifHor,
-                       x2=  1e-4,
-                       deltaX=  1e-5) + tmp1)/(1 + tmp1),
-        l=  1,
-        u=  8,
-        deltaX=  0.1));
   annotation (
     defaultComponentName="skyCle",
     Documentation(info="<html>
 <p>
 This component computes the sky clearness.
 </p>
+<h4>Implementation</h4>
+<p>
+In the <code>Annex60</code> library, <code>HGloHor</code>
+is always larger than <i>1E-4</i>,
+minus some small undershoot due to regularization. Hence,
+the implementation is not simplified for
+<code>HGloHor &lt; Modelica.Constants.small</code>.
+</p>
+<p>
+The regularization in the <code>Annex60.Utilities.Math.Functions.smoothMax</code>
+call is such that it usually is not trigger.
```

Comment:
This looks like a typo? This sentence does not seem grammatically correct to me.